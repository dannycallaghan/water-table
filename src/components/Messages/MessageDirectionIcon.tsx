function MessageDirectionIcon(props: { direction: string }) {
  if (props.direction.toLowerCase() === 'mo') {
    return (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        strokeWidth="2"
        stroke="currentColor"
        className="size-4 text-blue-600"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          d="M19.5 13.5 12 21m0 0-7.5-7.5M12 21V3"
        />
      </svg>
    );
  }
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      fill="none"
      viewBox="0 0 24 24"
      strokeWidth="2"
      stroke="currentColor"
      className="size-4 text-blue-600"
    >
      <path
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M4.5 10.5 12 3m0 0 7.5 7.5M12 3v18"
      />
    </svg>
  );
}

export default MessageDirectionIcon;
