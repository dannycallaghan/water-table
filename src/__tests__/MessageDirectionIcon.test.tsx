import { expect, test, describe } from 'vitest';
import { render } from '@testing-library/react';
import MessageDirectionIcon from '../components/Messages/MessageDirectionIcon.tsx';

describe('MessageDirectionIcon', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<MessageDirectionIcon direction={'MO'} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('renders correctly', () => {
    const { asFragment } = render(<MessageDirectionIcon direction={'MT'} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
