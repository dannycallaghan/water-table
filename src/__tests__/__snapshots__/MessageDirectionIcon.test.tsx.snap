// Vitest Snapshot v1

exports[`MessageDirectionIcon > renders correctly 1`] = `
<DocumentFragment>
  <svg
    class="size-4 text-blue-600"
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19.5 13.5 12 21m0 0-7.5-7.5M12 21V3"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</DocumentFragment>
`;

exports[`MessageDirectionIcon > renders correctly 2`] = `
<DocumentFragment>
  <svg
    class="size-4 text-blue-600"
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.5 10.5 12 3m0 0 7.5 7.5M12 3v18"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</DocumentFragment>
`;
